tendrl/commons/jobs/__init__.py

```diff
@@ -96,6 +111,17 @@ def run(self):
     def stop(self):
         pass
 
+    def read_job(self, key):
+        result = {}
+        job = tendrl_ns.etcd_orm.client.read(key)
+        if hasattr(job, '_children'):
+            for item in job._children:
+                if "dir" in item:
+                    result[item["key"].split("/")[-1]] =  self.read_job(item["key"])
+                else:
+                    result[item["key"].split("/")[-1]] = item["value"]
```

Comment:
I feel if you want to make this a generic utility, consider adding the recursive option to your function which facilitates tendrl_ns.etcd_orm.client.read(key, recursive=True) operation but yes the parsing logic might need to be enhanced..
Just a suggestion ...